src/vdaf/prio3.rs

```diff
@@ -913,10 +915,21 @@ where
             }
             let own_joint_rand_part = joint_rand_part_prg.into_seed();
 
-            let mut corrected_joint_rand_parts = public_share.joint_rand_parts.clone();
-            corrected_joint_rand_parts[agg_id as usize] = own_joint_rand_part.clone();
-
-            let joint_rand_seed = Self::derive_joint_rand_seed(corrected_joint_rand_parts.iter());
```

Comment:
It's hard to tell (for me anyway) if this this code is correct. To make this readable to someone who doesn't know rust that well or doesn't know the spec, a comment would be useful.

The goal here is produce an iterator over `public_share.joint_rand_parts` in which the `agg_id`-th part is replaced by the properly derived value.